### PR TITLE
Add support for running tests on alternate versions

### PR DESCRIPTION
**Pull request by Mothblocks**
Tested and working--includes a line that @AnturK will comment out to create an alternate test on 515.

### CODE REVIEW

File: tools/ci/install_byond.sh [hunk 1]
```diff
@@ -1,7 +1,10 @@
 #!/bin/bash
 set -euo pipefail
 
-source dependencies.sh
+# BYOND_MAJOR and BYOND_MINOR can be explicitly set, such as in alt_byond_versions.txt
+if [ -z "${BYOND_MAJOR+x}" ]; then
+  source dependencies.sh;
```
Comment (ZephyrTFA) [inline tools/ci/install_byond.sh:6]:
```suggestion
  source dependencies.sh
```

**Comment by MrStonedOne** [comment]
:heart:

**Comment by Mothblocks** [comment]
I think the conditional matrix is causing issues, I'll figure it out

**Comment by Mothblocks** [comment]
Hrm I think there is a bug in reusable workflows

**Comment by Mothblocks** [comment]
Yeah, there appears to be a bug in reusable workflows. It looks like we can't have alternate versions connect to screenshot tests? Not the biggest deal but very annoying

**Comment by Mothblocks** [comment]
Oh god no it's happening even without 🥲

**Comment by Mothblocks** [comment]
OH WAIT it's because of required checks. They're different now. My code worked fine this entire time

**Comment by Mothblocks** [comment]
The PR needs to be admin merged. It is expecting tests that were renamed (because they are now nested, it's automatic and I can't change it). Then after PR is merged we will need to update required checks to include the new tests. This has the added benefit of requiring people get from latest master.

When 515 is merged especially, and we start alternate testing on it, we can add that as a required check as well.

**Pull request merged**